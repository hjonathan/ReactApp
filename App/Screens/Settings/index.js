import {connect} from "react-redux";
import Settings from "./settings";
import actions from "../../Actions";

/**
 * Return the state form store to props of screen
 * @param state
 * @returns {{error: null, server: null}}
 */
const mapStateToProps = state => ({
    isLoggedIn: state.Oauth.isLoggedIn,
    error: state.Settings.error || null,
    server: state.Settings.server || null,
    progressShow: state.Net.progressShow,
    isConnected: state.Net.isConnected
});

/**
 * Map methods handlers with the component view
 * @param dispatch
 */
const mapDispatchToProps = dispatch => ({
    updateUrl: url => dispatch(actions.settings.updateUrl({url})),
    updateWorkspace: workspace => dispatch(actions.settings.updateWorkspace({workspace})),
    syncDown: () => dispatch(actions.net.syncDown.request())
});
export default connect(mapStateToProps, mapDispatchToProps)(Settings);
