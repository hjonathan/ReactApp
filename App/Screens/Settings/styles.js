import {StyleSheet} from "react-native";
import * as Res from "../../Assets/resources";

export default StyleSheet.create({
    mainContainer: {
        padding: 16
    },
    sectionTitle: {
        fontSize: 14,
        color: Res.colors.teal,
        fontWeight: "bold"
    },
    fieldContainer: {
        paddingVertical: 12
    },
    fieldTitle: {
        fontSize: 16,
        color: "#000"
    },
    divider: {
        width: "100%",
        height: 1,
        backgroundColor: "#ccc"
    },
    inputText: {
        color: "#000",
        fontSize: 16
    },
    disabled: {
        color: "#ccc"
    }
});
