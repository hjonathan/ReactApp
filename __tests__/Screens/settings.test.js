import React from 'react'
import {shallow} from 'enzyme';
import ConnectSettings from '../../App/Screens/Settings';
import Settings from '../../App/Screens/Settings/settings';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

jest.mock('react-native-restart', () => require.requireActual('../../__mocks__/react-native-restart').default);

describe('Screen >>> Settings', () => {
    const initState = {
        Settings: {
            error: {
                error: "",
                error_description: ""
            },
            server: {
                url: "https://poc-1.processmaker.net",
                workspace: "workflowpoc1"
            }
        },
        Net: {
            progressShow: false
        },
        Oauth: {
            isLoggedIn: true
        }
    };
    const initStateEmpty = {
        Settings: {
            error: null,
            server: null
        },
        Net: {
            progressShow: false
        },
        Oauth: {
            isLoggedIn: true
        }
    };
    const propTypes = {
        updateUrl: () => {
        },
        updateWorkspace: () => {
        },
        syncDown: () => {
        }
    };
    const navigation = {
        dispatch: () => {},
        setParams: () => {}
    };
    const toast = {
        _toastSettings: {
            show: (message, duration) => {}
        }
    };
    const nextProps = {
        error: {
            error_description: "message error"
        },
        server: {
            url: "https://poc-1.processmaker.net",
            workspace: "workflowpoc1"
        }
    };
    const nextPropsElse = {
        error: null,
        server: {
            url: "https://poc-1.processmaker.net",
            workspace: "workflowpoc1"
        }
    };
    const mockStore = configureStore();
    let connect,
        connectEmpty,
        view;

    beforeEach(() => {
        connect = shallow(<ConnectSettings store={mockStore(initState)}/>);
        connectEmpty = shallow(<ConnectSettings store={mockStore(initStateEmpty)}/>);
        view = shallow(<Settings
            isLoggedIn={true}
            navigation={navigation}
            server={initState.Settings.server}
            updateUrl={propTypes.updateUrl}
            updateWorkspace={propTypes.updateWorkspace}
            syncDown={propTypes.syncDown} />);
    });

    it('Settings connect', () => {
        expect(toJson(connect)).toMatchSnapshot();
    });

    it('mapStateToProps', () => {
        expect(toJson(connectEmpty)).toMatchSnapshot();
    });

    it('Check static navigationOptions', () => {
        expect(Settings.navigationOptions({navigation}).title)
            .toEqual("menu_drawer_settings_item_settings");
        expect(Settings.navigationOptions({navigation}).headerTintColor)
            .toEqual("#FFF");
        expect(Settings.navigationOptions({navigation}).headerStyle)
            .toEqual({backgroundColor: "#1F6CBC"});
        expect(Settings.navigationOptions({navigation}).headerLeft).toMatchSnapshot();
    });

    it('mapDispatchToProps', () => {
        expect(connect.props('children').updateUrl()).toMatchSnapshot();
        expect(connect.props('children').updateWorkspace()).toMatchSnapshot();
        expect(connect.props('children').syncDown()).toMatchSnapshot();
    });

    it('Render Settings view', () => {
        expect(toJson(view)).toMatchSnapshot();
    });

    it('OnPress sections', () => {
        let mainContainer = view.props('children').children.props.children.props.children.props.children,
            serverUrl = mainContainer[0].props.children[1].props,
            workspace = mainContainer[0].props.children[3].props,
            cleanCache = mainContainer[1].props.children[1].props,
            externalLibs = mainContainer[1].props.children[3].props.children[0].props,
            languages = mainContainer[2].props.children[1].props;

        expect(toJson(serverUrl.onPress())).toMatchSnapshot();
        expect(toJson(workspace.onPress())).toMatchSnapshot();
        expect(toJson(cleanCache.onPress())).toMatchSnapshot();
        expect(toJson(externalLibs.onPress())).toMatchSnapshot();
        expect(toJson(languages.onPress())).toMatchSnapshot();
        expect(view.instance()._syncDown()).toMatchSnapshot()
    });


    it('componentWillReceiveProps function', () => {
        expect(view.instance().componentWillReceiveProps(nextPropsElse)).toMatchSnapshot();
        view.instance().refs = toast;
        expect(view.instance().componentWillReceiveProps(nextProps)).toMatchSnapshot();
    });

    it('updateServer function', () => {
        view.instance().setState({tmpServer: ""});
        view.instance().refs = toast;
        expect(view.instance().updateServer()).toMatchSnapshot();
    });

    it('initialSettings function', async () => {
        expect(await view.instance().initialSettings()).toMatchSnapshot();
    });

    it('_onChangeServerUrl function', () => {
        expect(view.instance()._onChangeServerUrl("http://sample.com")).toMatchSnapshot();
        expect(view.instance()._onChangeServerUrl("")).toMatchSnapshot();
    });

    it('_onChangeWorkspace function', () => {
        expect(view.instance()._onChangeWorkspace("")).toMatchSnapshot();
        expect(view.instance()._onChangeWorkspace("sample")).toMatchSnapshot();
    });

    it('_onChangeExternalLibs function', () => {
        expect(view.instance()._onChangeExternalLibs(true)).toMatchSnapshot();
    });

    it('_onChangeLanguage function', () => {
        expect(view.instance()._onSelectLanguage("ar")).toMatchSnapshot();
        expect(view.instance()._onSelectLanguage()).toMatchSnapshot();
    });

    it('renderDialog ServerUrl function', () => {
        const dialogServerUrl = view.instance()._renderDialogServerUrl(),
            controls = dialogServerUrl.props.children.props.children[1].props,
            buttonCancel = controls.children[0],
            buttonOk = controls.children[1];
        expect(dialogServerUrl.props.onTouchOutside()).toMatchSnapshot();
        expect(buttonCancel.props.onPress()).toMatchSnapshot();
        view.instance().refs = toast;
        expect(buttonOk.props.onPress()).toMatchSnapshot();
    });

    it('renderDialog ServerUrl function', () => {
        const dialogServerUrl = view.instance()._renderDialogServerUrl(),
            controls = dialogServerUrl.props.children.props.children[1].props,
            buttonOk = controls.children[1];
        view.instance().setState({tmpServer: "http://example.com"});
        expect(buttonOk.props.onPress()).toMatchSnapshot();
    });

    it('renderDialog Workspace function', () => {
        const dialogWorkspace = view.instance()._renderDialogWorkspace(),
            controls = dialogWorkspace.props.children.props.children[1].props,
            buttonCancel = controls.children[0],
            buttonOk = controls.children[1];
        expect(dialogWorkspace.props.onTouchOutside()).toMatchSnapshot();
        expect(buttonCancel.props.onPress()).toMatchSnapshot();
        expect(buttonOk.props.onPress()).toMatchSnapshot();
    });

    it('renderDialog CleanCache function', () => {
        let dialogCleanCache = view.instance()._renderDialogCleanCache(),
            controls = dialogCleanCache.props.children.props.children[1].props,
            buttonCancel = controls.children[0],
            buttonOk = controls.children[1];
        expect(dialogCleanCache.props.onTouchOutside()).toMatchSnapshot();
        expect(buttonCancel.props.onPress()).toMatchSnapshot();
        expect(buttonOk.props.onPress()).toMatchSnapshot();
    });

    it('renderDialog Languages function', () => {
        let dialogLanguages = view.instance()._renderDialogLanguages(),
            languages = dialogLanguages.props.children.props.children[0],
            firstLanguage = languages[0],
            controls = dialogLanguages.props.children.props.children[1].props,
            buttonCancel = controls.children;
        expect(dialogLanguages.props.onTouchOutside()).toMatchSnapshot();
        expect(buttonCancel.props.onPress()).toMatchSnapshot();
        expect(firstLanguage.props.onPress()).toMatchSnapshot();
    });
});
