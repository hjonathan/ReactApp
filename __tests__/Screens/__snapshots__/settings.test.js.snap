// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen >>> Settings _onChangeExternalLibs function 1`] = `undefined`;

exports[`Screen >>> Settings _onChangeLanguage function 1`] = `undefined`;

exports[`Screen >>> Settings _onChangeLanguage function 2`] = `undefined`;

exports[`Screen >>> Settings _onChangeServerUrl function 1`] = `undefined`;

exports[`Screen >>> Settings _onChangeServerUrl function 2`] = `undefined`;

exports[`Screen >>> Settings _onChangeWorkspace function 1`] = `undefined`;

exports[`Screen >>> Settings _onChangeWorkspace function 2`] = `undefined`;

exports[`Screen >>> Settings Check static navigationOptions 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "padding": 16,
    }
  }
>
  <Icon
    allowFontScaling={false}
    name="arrow-back"
    size={12}
    style={
      Object {
        "color": "#FFF",
        "fontSize": 24,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Screen >>> Settings OnPress sections 1`] = `undefined`;

exports[`Screen >>> Settings OnPress sections 2`] = `undefined`;

exports[`Screen >>> Settings OnPress sections 3`] = `undefined`;

exports[`Screen >>> Settings OnPress sections 4`] = `undefined`;

exports[`Screen >>> Settings OnPress sections 5`] = `undefined`;

exports[`Screen >>> Settings OnPress sections 6`] = `undefined`;

exports[`Screen >>> Settings Render Settings view 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
  >
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Component
        style={
          Object {
            "padding": 16,
          }
        }
      >
        <Component
          style={
            Object {
              "paddingBottom": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#009688",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            preference_end_point_settings
          </Text>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[Function]}
            underlayColor="#E1E3E8"
          >
            <Component
              style={
                Object {
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
              >
                preference_end_point_chose
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </Component>
          </TouchableHighlight>
          <Component
            style={
              Object {
                "backgroundColor": "#ccc",
                "height": 1,
                "width": "100%",
              }
            }
          />
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[Function]}
            underlayColor="#E1E3E8"
          >
            <Component
              style={
                Object {
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
              >
                preference_end_point_name
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </Component>
          </TouchableHighlight>
        </Component>
        <Component
          style={
            Object {
              "paddingBottom": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#009688",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            preference_erasercache_Title
          </Text>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[Function]}
            underlayColor="#E1E3E8"
          >
            <Component
              style={
                Object {
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
              >
                preference_eraser_sumary
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                preference_erasercache_Title_Clean
              </Text>
            </Component>
          </TouchableHighlight>
          <Component
            style={
              Object {
                "backgroundColor": "#ccc",
                "height": 1,
                "width": "100%",
              }
            }
          />
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPress={[Function]}
              style={
                Object {
                  "width": "85%",
                }
              }
              underlayColor="#E1E3E8"
            >
              <Component
                style={
                  Object {
                    "paddingVertical": 12,
                  }
                }
              >
                <Component
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <Component
                    style={
                      Object {
                        "flex": 1,
                        "flexWrap": "wrap",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 16,
                        }
                      }
                    >
                      preference_enable_external_libraries_cache
                    </Text>
                  </Component>
                </Component>
              </Component>
            </TouchableHighlight>
            <Switch
              disabled={false}
              onValueChange={[Function]}
              style={
                Object {
                  "width": "15%",
                }
              }
              value={true}
            />
          </Component>
        </Component>
        <Component
          style={
            Object {
              "paddingBottom": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#009688",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            preference_language_selection_title
          </Text>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[Function]}
            underlayColor="#E1E3E8"
          >
            <Component
              style={
                Object {
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                English
              </Text>
            </Component>
          </TouchableHighlight>
        </Component>
        <Component
          style={
            Object {
              "paddingBottom": 16,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#009688",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            preference_offline_settings
          </Text>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPress={[Function]}
            underlayColor="#E1E3E8"
          >
            <Component
              style={
                Object {
                  "paddingVertical": 12,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
              >
                preference_offline_download_data
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                preference_offline_data_downloaded
              </Text>
            </Component>
          </TouchableHighlight>
        </Component>
        <Dialog
          onRequestClose={[Function]}
          onTouchOutside={[Function]}
          title="preference_end_point_chose"
          visible={false}
        >
          <Component
            style={
              Object {
                "flexDirection": "column",
                "margin": -8,
              }
            }
          >
            <Component>
              <FormInput
                autoFocus={true}
                defaultValue=""
                inputStyle={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
                normalizeFontSize={true}
                onChangeText={[Function]}
                underlineColorAndroid="#009688"
              />
            </Component>
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingVertical": 8,
                }
              }
            >
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  CANCEL
                </Text>
              </TouchableHighlight>
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  OK
                </Text>
              </TouchableHighlight>
            </Component>
          </Component>
        </Dialog>
        <Dialog
          onRequestClose={[Function]}
          onTouchOutside={[Function]}
          title="preference_end_point_name"
          visible={false}
        >
          <Component
            style={
              Object {
                "flexDirection": "column",
                "margin": -8,
              }
            }
          >
            <Component>
              <FormInput
                autoFocus={true}
                defaultValue=""
                inputStyle={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
                normalizeFontSize={true}
                onChangeText={[Function]}
                underlineColorAndroid="#009688"
              />
            </Component>
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingVertical": 8,
                }
              }
            >
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  CANCEL
                </Text>
              </TouchableHighlight>
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  OK
                </Text>
              </TouchableHighlight>
            </Component>
          </Component>
        </Dialog>
        <Dialog
          onRequestClose={[Function]}
          onTouchOutside={[Function]}
          title="Cache_Erased"
          visible={false}
        >
          <Component
            style={
              Object {
                "flexDirection": "column",
                "margin": -8,
              }
            }
          >
            <Component>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                  }
                }
              >
                Message_Cache_Erased
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingVertical": 8,
                }
              }
            >
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  CANCEL
                </Text>
              </TouchableHighlight>
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  OK
                </Text>
              </TouchableHighlight>
            </Component>
          </Component>
        </Dialog>
        <Dialog
          onRequestClose={[Function]}
          onTouchOutside={[Function]}
          visible={false}
        >
          <Component
            style={
              Object {
                "flexDirection": "column",
                "margin": -8,
              }
            }
          >
            <CheckBox
              center={false}
              checked={true}
              checkedColor="#009688"
              checkedIcon="radio-button-checked"
              containerStyle={
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "padding": 0,
                }
              }
              iconRight={false}
              iconType="material"
              key="en"
              onPress={[Function]}
              right={false}
              size={24}
              textStyle={
                Object {
                  "fontWeight": "normal",
                }
              }
              title="English"
              uncheckedColor="#bfbfbf"
              uncheckedIcon="radio-button-unchecked"
            />
            <CheckBox
              center={false}
              checked={false}
              checkedColor="#009688"
              checkedIcon="radio-button-checked"
              containerStyle={
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "padding": 0,
                }
              }
              iconRight={false}
              iconType="material"
              key="es"
              onPress={[Function]}
              right={false}
              size={24}
              textStyle={
                Object {
                  "fontWeight": "normal",
                }
              }
              title="Español"
              uncheckedColor="#bfbfbf"
              uncheckedIcon="radio-button-unchecked"
            />
            <CheckBox
              center={false}
              checked={false}
              checkedColor="#009688"
              checkedIcon="radio-button-checked"
              containerStyle={
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "padding": 0,
                }
              }
              iconRight={false}
              iconType="material"
              key="fr"
              onPress={[Function]}
              right={false}
              size={24}
              textStyle={
                Object {
                  "fontWeight": "normal",
                }
              }
              title="Français"
              uncheckedColor="#bfbfbf"
              uncheckedIcon="radio-button-unchecked"
            />
            <CheckBox
              center={false}
              checked={false}
              checkedColor="#009688"
              checkedIcon="radio-button-checked"
              containerStyle={
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "padding": 0,
                }
              }
              iconRight={false}
              iconType="material"
              key="pt"
              onPress={[Function]}
              right={false}
              size={24}
              textStyle={
                Object {
                  "fontWeight": "normal",
                }
              }
              title="Portugues"
              uncheckedColor="#bfbfbf"
              uncheckedIcon="radio-button-unchecked"
            />
            <CheckBox
              center={false}
              checked={false}
              checkedColor="#009688"
              checkedIcon="radio-button-checked"
              containerStyle={
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                  "padding": 0,
                }
              }
              iconRight={false}
              iconType="material"
              key="ar"
              onPress={[Function]}
              right={false}
              size={24}
              textStyle={
                Object {
                  "fontWeight": "normal",
                }
              }
              title="العربية"
              uncheckedColor="#bfbfbf"
              uncheckedIcon="radio-button-unchecked"
            />
            <Component
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingVertical": 8,
                }
              }
            >
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPress={[Function]}
                style={
                  Object {
                    "paddingHorizontal": 16,
                    "paddingVertical": 8,
                  }
                }
                underlayColor="#E1E3E8"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#009688",
                      "fontWeight": "bold",
                    }
                  }
                >
                  CANCEL
                </Text>
              </TouchableHighlight>
            </Component>
          </Component>
        </Dialog>
        <Toast
          fadeInDuration={500}
          fadeOutDuration={500}
          opacity={1}
          position="bottom"
          positionValue={200}
          style={
            Object {
              "backgroundColor": "#3397E1",
              "marginHorizontal": 24,
            }
          }
          textStyle={
            Object {
              "color": "white",
              "textAlign": "center",
            }
          }
        />
        <ProgressDialog
          message="login_dialog_loading_description_setting_up_process_maker"
          title="login_dialog_loading_title"
          visible={false}
        />
        <ProgressDialog
          message="preference_sync_pmdynaforms_description"
          title="preference_sync_pmdynaforms_title"
        />
      </Component>
    </Component>
  </ScrollView>
</Component>
`;

exports[`Screen >>> Settings Settings connect 1`] = `
<Settings
  error={
    Object {
      "error": "",
      "error_description": "",
    }
  }
  isLoggedIn={true}
  progressShow={false}
  server={
    Object {
      "url": "https://poc-1.processmaker.net",
      "workspace": "workflowpoc1",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  syncDown={[Function]}
  updateUrl={[Function]}
  updateWorkspace={[Function]}
/>
`;

exports[`Screen >>> Settings componentWillReceiveProps function 1`] = `undefined`;

exports[`Screen >>> Settings componentWillReceiveProps function 2`] = `undefined`;

exports[`Screen >>> Settings initialSettings function 1`] = `undefined`;

exports[`Screen >>> Settings mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "url": undefined,
  },
  "type": "settings/updateUrl",
}
`;

exports[`Screen >>> Settings mapDispatchToProps 2`] = `
Object {
  "payload": Object {
    "workspace": undefined,
  },
  "type": "settings/updateWorkspace",
}
`;

exports[`Screen >>> Settings mapDispatchToProps 3`] = `
Object {
  "payload": Object {
    "status": undefined,
  },
  "type": "net/syncDown/request",
}
`;

exports[`Screen >>> Settings mapStateToProps 1`] = `
<Settings
  error={null}
  isLoggedIn={true}
  progressShow={false}
  server={null}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  syncDown={[Function]}
  updateUrl={[Function]}
  updateWorkspace={[Function]}
/>
`;

exports[`Screen >>> Settings renderDialog CleanCache function 1`] = `undefined`;

exports[`Screen >>> Settings renderDialog CleanCache function 2`] = `undefined`;

exports[`Screen >>> Settings renderDialog CleanCache function 3`] = `undefined`;

exports[`Screen >>> Settings renderDialog Languages function 1`] = `undefined`;

exports[`Screen >>> Settings renderDialog Languages function 2`] = `undefined`;

exports[`Screen >>> Settings renderDialog Languages function 3`] = `undefined`;

exports[`Screen >>> Settings renderDialog ServerUrl function 1`] = `undefined`;

exports[`Screen >>> Settings renderDialog ServerUrl function 2`] = `undefined`;

exports[`Screen >>> Settings renderDialog ServerUrl function 3`] = `undefined`;

exports[`Screen >>> Settings renderDialog ServerUrl function 4`] = `undefined`;

exports[`Screen >>> Settings renderDialog Workspace function 1`] = `undefined`;

exports[`Screen >>> Settings renderDialog Workspace function 2`] = `undefined`;

exports[`Screen >>> Settings renderDialog Workspace function 3`] = `undefined`;

exports[`Screen >>> Settings updateServer function 1`] = `undefined`;
